### ISSUE TITLE
Reproduce report results on ADE20K failed in segmentation task.

### ISSUE THREAD

**Issue by leepms** [EdwardChasel/BinaryAttention#3]
Thanks for your amazing work!

I am currently trying to apply BinaryAttention to a semantic segmentation task (ADE20K) using the OffSeg model. However, I encountered an issue where it does not work properly during validation.

To adapt your method, I reimplemented the self-distillation framework based on your code. Specifically, I replaced the BEC loss used in classification with a standard cross-entropy (CE) loss as the base loss, and adopted a hard distillation strategy, where the student model is trained to match the teacher’s outputs using CE loss.

Since the segmentation code mentioned in the paper has not been released yet, I would greatly appreciate any suggestions on how to properly adapt BinaryAttention to segmentation tasks.

Also, do you have plans to release the full implementation for segmentation in the near future? or would you share more details in your experiemnts?

Thank you very much!

**Comment by EdwardChasel** [comment]
Thanks for your attention. We conducted the semantic segmentation task (ADE20K) using [MMSegmentation](https://github.com/open-mmlab/mmsegmentation/tree/v0.20.2). In the downstream experiments, we fine-tuned the models with the pre-trained backbone weights and did not adopt self-distillation strategies. For your convenience, you can refer to this [codebase](https://github.com/baaivision/EVA/tree/master/EVA-02/seg).

**Comment by leepms** [comment]
Thanks your answers! Your methods actually make sense for in my model only introduce little degrade in mIoU and mAcc metrics, But It should take proper lr setting to avoid quickly overfitting on train data.

In my experiment, lr=1e-7 per sample is proper and about converge in 15 epoches.
